### PR TITLE
System.data.common dotnet5.1

### PR DESCRIPTION
**Pull request by ericstj**
System.Data.Common needs to support dotnet5.1 & net45 but was
currently targeting dotnet5.4 and net46.

This fixes both the portable implementation and the facade and merges
the facade into the main project.

/cc @weshaggard @saurabh500 @chcosta

### CODE REVIEW

File: src/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs [hunk 1]
```diff
@@ -464,17 +466,18 @@ private static Hashtable SplitConnectionString(string connectionString, Hashtabl
...
+                foreach (KeyValuePair<string, string> entry in parsedvalues)
                 {
-                    string keyname = (string)entry.Key;
-                    string value1 = (string)entry.Value;
-                    string value2 = (string)parsetable[keyname];
-                    Debug.Assert(parsetable.Contains(keyname), "ParseInternal code vs. regex mismatch keyname <" + keyname + ">");
+                    string keyname = entry.Key;
+                    string value1 = entry.Value;
+                    string value2 = null;
+                    parsetable.TryGetValue(keyname, out value2);
```
Comment (weshaggard) [inline src/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs]: Why are you switching to TryGetValue? Does the index throw in dictionary but not in hashtable?

File: src/System.Data.Common/src/project.json [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+        "System.Resources.ResourceManager": "4.0.0",
+        "System.Runtime": "4.0.0",
+        "System.Runtime.Extensions": "4.0.0",
+        "System.Text.RegularExpressions": "4.0.0",
+        "System.Threading.Tasks": "4.0.0"
+      }
+    },
+    "net45": {
+      "dependencies": {
+        "Microsoft.TargetingPack.NETFramework.v4.5": "1.0.1"
```
Comment (weshaggard) [inline src/System.Data.Common/src/project.json]: I would expect this to fail package validation (https://github.com/dotnet/corefx/blob/master/dir.props#L79) is it not failing?

**Comment by weshaggard** [comment]
LGTM

File: src/System.Data.Common/src/project.json (hunk 2 above)
Comment (dagood) [inline src/System.Data.Common/src/project.json]: Yeah, I noticed this just a bit ago when I was trying to use that rule to upgrade the targetingpack dependencies. In my (TFS) changes I've included `(?i)` near the beginning of each rule, although we could also fix this by changing the regex mode in buildtools.

File: src/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs (hunk 1 above)
Comment (ericstj) [inline src/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs]: Yeah, hash table returns null.

Comment (ericstj) [inline src/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs]: And dictionary throws

File: src/System.Data.Common/src/project.json (hunk 2 above)
Comment (ericstj) [inline src/System.Data.Common/src/project.json]: Ill fix that version.  Why shouldn't I be using 1.0.1, by the way?  Do you want those to be deleted from the feed?

Comment (weshaggard) [inline src/System.Data.Common/src/project.json]: They should be version 1.0.1 but that was coming as part of @dagood's change but I'm cool whenever we want to update them independently we should just try to do it holistically.

**Pull request merged**